docs/source/conf.py

```diff
@@ -160,51 +165,154 @@ def run_apidoc(_):
     options = [
         package_dir,
         *exclude_api_patterns,
-        '-o', apidoc_dir,
+        '-o',
+        apidoc_dir,
         '--private',
         '--force',
         '--no-headings',
         '--module-first',
         '--no-toc',
-        '--maxdepth', '4',
+        '--maxdepth',
+        '4',
     ]
 
     # See https://stackoverflow.com/a/30144019
     env = os.environ.copy()
-    env['SPHINX_APIDOC_OPTIONS'] = 'members,special-members,private-members,undoc-members,show-inheritance'
+    env['SPHINX_APIDOC_OPTIONS'
+        ] = 'members,special-members,private-members,undoc-members,show-inheritance,ignore-module-all'
     subprocess.check_call([cmd_path] + options, env=env)
 
-def setup(app):
-    if os.environ.get('RUN_APIDOC', None) != 'False':
-        app.connect('builder-inited', run_apidoc)
 
 # Warnings to ignore when using the -n (nitpicky) option
 with open('nitpick-exceptions', 'r') as handle:
     nitpick_ignore = [
         tuple(line.strip().split(None, 1)) for line in handle.readlines() if line.strip() and not line.startswith('#')
     ]
+# ignore anything ending in `Type`, since it is likely a `typing.TypeVar`, which sphinx-apidoc doesn't handle
+nitpick_ignore_regex = [('py:.*', '.+Type')]
```

Comment:
Hmm, this also includes all our custom `ParameterType` implementations in `aiida.cmdline.params.types`. Don't think it is the worst, but is there a way we can exclude those?

Reply:
This just stops a warning from being emitted if it can't resolve the reference, it won't stop the reference from being resolved if present

Reply:
Yeah I get that, but if there is warning related to our param types, that is probably because there is a real problem, and this way it will go unnoticed.

Reply:
Eurgh, unless you are literally going to be using ``:py:obj:`aiida.cmdline.params.types.WrongType` `` in the documentation, or a docstring, I don't think it's going to be a big problem. Obviously, actual typing errors should be tested by mypy

Reply:
But that is the whole point of these warnings, to warn you if you (by accident) type a wrong resource, for example a typo.

Reply:
There are literally no sphinx references to `ParamType` anywhere in the documentation/docstrings, but anyhow I've removed this and reinstated a lovely long `nitpick_exceptions` list.